core/src/main/java/com/novoda/noplayer/AdvertsLoader.java

```diff
@@ -16,7 +16,7 @@
     interface Callback {
         void onAdvertsLoaded(List<AdvertBreak> advertBreaks);
 
-        void onAdvertsError(String message);
+        void onAdvertsError(Exception cause);
```

Comment:
Made more sense in the end not to limit the information for an error to a `String`, will help our clients diagnose issues if they occur.